mqtt/mqtt-broker/src/broker.rs

```diff
@@ -355,6 +362,9 @@ where
             debug!("no session for {}", client_id);
         }
         debug!("drop connection handled.");
+
+        self.notify_disconnect(&client_id).await;
```

Comment:
do we need to do this only if session was found and disconnected (ln:354)?

Reply:
probably, but I erred on the side of caution. I'd rather send extra messages than miss one.